### PR TITLE
Update DG for `filter` command

### PR DESCRIPTION
**Pull request by niekis**

### CODE REVIEW

File: docs/diagrams/FilterSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+create CommandResult
+FilterCommand -> CommandResult
+activate CommandResult
+
+CommandResult --> FilterCommand
+deactivate CommandResult
+
+FilterCommand --> LogicManager : result
+deactivate FilterCommand
```
Comment (rexcyrio) [inline docs/diagrams/FilterSequenceDiagram.puml:104]: I think there should be a `destroy FilterCommand` here? (plus hidden arrow of course)

File: docs/diagrams/FilterSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+participant "fc:FilterCommand" as FilterCommand LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "fp :FieldContainsPartialKeywordsPredicate" as FieldContainsPartialKeywordsPredicate MODEL_COLOR
+participant ":Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("filter e/gmail r/3sg")
```
Comment (rexcyrio) [inline docs/diagrams/FilterSequenceDiagram.puml:18]: I believe the current app only allows uppercase ranks. Maybe we should stick with `r/3SG` in the DG until we fix it?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -275,6 +275,31 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 *{more aspects and alternatives to be added}*
 
+### Filter feature
```
Comment (rexcyrio) [inline docs/DeveloperGuide.md:278]: Is the table of contents updated?

File: docs/diagrams/FilterSequenceDiagram.puml (hunk 1 above)
Comment (niekis) [reply docs/diagrams/FilterSequenceDiagram.puml:104]: okie thanks!

File: docs/diagrams/FilterSequenceDiagram.puml (hunk 2 above)
Comment (niekis) [reply docs/diagrams/FilterSequenceDiagram.puml:18]: oh, not for filter! I wanted to reiterate the fact that it isn't limited by the same constraints as the field values

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (niekis) [reply docs/DeveloperGuide.md:278]: oh good eye, will fix it!

**Review by minosx31 — APPROVED** [review]
LGTM! 🔥

File: docs/diagrams/FilterSequenceDiagram.puml (hunk 2 above)
Comment (rexcyrio) [reply docs/diagrams/FilterSequenceDiagram.puml:18]: oh ya you're right :fire:

**Review by rexcyrio — APPROVED** [review]
LGTM :+1:

**Review by jugheadjones10 — APPROVED** [review]

**Pull request merged**